runtime/syntax/dockerfile.vim

```diff
@@ -1,26 +1,45 @@
 " dockerfile.vim - Syntax highlighting for Dockerfiles
 " Maintainer:   Honza Pokorny <https://honza.ca>
-" Version:      0.6
-" Last Change:  2019 Aug 16
 " License:      BSD
 
+" https://docs.docker.com/engine/reference/builder/
 
 if exists("b:current_syntax")
     finish
 endif
 
-let b:current_syntax = "dockerfile"
+syntax include @JSON syntax/json.vim
+unlet b:current_syntax
+
+syntax include @Shell syntax/sh.vim
+unlet b:current_syntax
 
 syntax case ignore
+syntax match dockerfileLinePrefix /\v^\s*(ONBUILD\s+)?\ze\S/ contains=dockerfileKeyword nextgroup=dockerfileInstruction skipwhite
+syntax region dockerfileFrom matchgroup=dockerfileKeyword start=/\v^\s*(FROM)\ze(\s|$)/ skip=/\v\\\_./ end=/\v((^|\s)AS(\s|$)|$)/ contains=dockerfileOption
```

Comment:
You can use `\<` or `\>` to match beginning or ending of a word. (`<` or `>` with `\v`.)
E.g. `start=/\v^\s*FROM>/`, `end=/\v(<AS>|$)/`

Reply:
That does read better but it allows some text that I don't want. For example, `end=/\v(<AS>|$)/` will match ".AS" or "%AS".

Does the word boundary perform significantly better? I'm willing to accept some incorrect edge cases if it avoids a toxic pattern.